File: swix/speed/speed.jl
function ndgrid{T}(v1::AbstractVector{T}, v2::AbstractVector{T})
    m, n = length(v1), length(v2)
    v1 = reshape(v1, m, 1)
    v2 = reshape(v2, 1, n)
    (repmat(v1, 1, n), repmat(v2, m, 1))
end


function pe1()
    N = 1e6
    sum = 0
    for i=1:N
        if i % 3 == 0 || i % 5 == 0
            sum += i
        end
    end
end

function pe10()
    N = 2e6
    primes = [1:N]
    for i in 2:sqrt(N)
        j = (2:N/i)*i
        primes[j] = 0
    end
end

function pe73()
    N = 1e3
    di = [1:N]
    ni = [1:N]
    (n, d) = ndgrid(ni, di)

    f = n ./ d

    f = unique(f)
    f = sort(f)
    f = f[f .< 1/2]
    f = f[f .> 1/3]
end

function soft_threshold()
    N = 1e2
    j = linspace(-1.0, 1.0, convert(Int64, N))
    x, y = ndgrid(j, j)
    z = x.^2 + y.^2
    i = abs(z) .< 0.5
    z[i] = 0
    z[!i] -= 0.5
end

function pi_approx()
    N = 1e6
    pie = 1.0
    for k in 1:N
        pie += (-1)^k / (2*k + 1)
    end
end

# timing
pe1()
pe10()
pe73()
soft_threshold()
pi_approx()

print("pe1: ")
println(@elapsed pe1())
print("pe10: ")
println(@elapsed pe10())
print("pe73: ")
println(@elapsed pe73())
print("st_time: ")
println(@elapsed soft_threshold())
print("pi_approx: ")
println(@elapsed pi_approx())

File: swix/speed/speed.py
from __future__ import division
from pylab import *
from timeit import timeit



def pe1():
    N = 1e6
    x = arange(N)
    i = argwhere((abs(x%3) < 1e-9) * (abs(x%5) < 1e-9))
def pe10():
    N = 2e6
    primes = arange(N)
    for i in arange(2, sqrt(N)):
        j = arange(2, N/i) * i
        j = asarray(j, dtype=int)
        primes[j] *= 0.0
def pe73():
    N = 1e3
    d = arange(N)+1
    n = arange(N)+1
    n, d = meshgrid(n, d)

    # setting the upper triangular to 0
    #il1 = triu_indices(n.shape[0])
    #n[il1] = 0

    f = n / d
    f = unique(f)

    i = (f > 1/3) & (f < 1/2)
def soft_threshold():
    N = 1e2
    j = linspace(-1, 1, num=N)
    (x, y) = meshgrid(j, j)
    z = pow(x, 2) + pow(y, 2)
    i = abs(z) < 0.5
    z[argwhere(i)] *= 0
    z[argwhere(~i)] -= 0.5
def pi_approx():
    N = 1e6
    k = arange(N, dtype=int)
    pi_approx = 1 / (2*k + 1)
    pi_approx[2*k[:N/2]+1] *= -1

print "pe1_time       : ", timeit(pe1 ,number=10)
print "pe10_time      : ", timeit(pe10 ,number=3)
print "pe73_time      : ", timeit(pe73 ,number=1)
print "st_time        : ", timeit(soft_threshold ,number=3)
print "pi_approx time : ", timeit(pi_approx ,number=10)

